### PR TITLE
ENH: Launcher

### LINKED ISSUE #7: Conda Recipe - Missing Dependencies
The conda recipe is missing dependencies such as ophyd, typhon and others...
Make the package be `noarch`...

### PR DESCRIPTION
**Pull request by hhslepicka**
This PR adds a luncher to LUCID and a demo folder.

- For now the launcher takes a mandatory argument `beamline` which will be used to query Happi for the devices to be added to the home screen.
- In order to allow for quick testing one can pass `beamline` as `DEMO` and it will launch LUCID with  a random number of fake devices.
- The demo folder contains an example of a `toolbar file` (which will be implemented in a following PR) as well as a Happi sample database and configuration files. That is useful if we want to test the whole scenario in which we load from Happi and generate the UI.

Closes #7

### CODE REVIEW

**Comment by hhslepicka** [comment]
Kicking Travis now that we have PyQtAds at conda-forge

File: lucid/launcher.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        required=False
+    )
+    parser.add_argument(
+        '--log_level',
+        help='Configure level of log display',
+        choices=['DEBUG', 'INFO', 'WARNING', 'ERROR', 'CRITICAL'],
+        default='INFO'
+    )
+
+    lucid_args = parser.parse_args()
```
Comment (klauer) [inline lucid/launcher.py]: I really like the pattern of making the arg parsing separate from the actual functionality, such that you can programmatically use the launcher without having to make up your own list of string arguments.
e.g.,
```
    args = top_parser.parse_args()
    kwargs = vars(args)
    launcher(**kwargs)
```

where `kwargs` are `{'version', 'beamline', 'toolbar', ...}`. Though you might consider popping `version` out first and handling it specially.

File: demo/launch.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+CWD=`pwd`
+CONFIG=${CWD}/happi_demo.cfg
+TOOLBAR=${CWD}/demo_toolbar.yml
```
Comment (klauer) [inline demo/launch.sh]: Found this while trying to run the example:
```suggestion
TOOLBAR=${CWD}/example_toolbar.yml
```

File: lucid/launcher.py (hunk 1 above)
Comment (hhslepicka) [reply lucid/launcher.py]: I like this idea! It sounds really good! The version I think will be handled by the parse_args special version case, right?

File: demo/launch.sh (hunk 2 above)
Comment (hhslepicka) [reply demo/launch.sh]: Good catch...

**Comment by hhslepicka** [comment]
I think we can iterate at this one until we think it is good... for sure it is a first stab at the problem but if we could at least make it a nice first stab it would be good! ;) So I am fine with keep working and addressing the comments in this PR for now...

**Comment by hhslepicka** [comment]
BTW, what an honor of receiving a `Hooray` from the ghost of @teddyrendahl 😆

**Comment by klauer** [comment]
After some discussion, I pushed my changes. We'll have to merge fixes from #6 on top of this.

**Comment by hhslepicka** [comment]
@klauer says: `Go for it.`

File: lucid/launcher.py (hunk 1 above)
Comment (klauer) [reply lucid/launcher.py]: I just learned about that option now, thanks!

**Pull request merged**